### PR TITLE
DCS-706 Moving to new get video bookings endpoint and add deletion links

### PR DESCRIPTION
**Pull request by andrewrlee**

### CODE REVIEW

File: tsconfig.json [hunk 1]
```diff
@@ -1,7 +1,7 @@
 {
   "compileOnSave": true,
   "compilerOptions": {
-    "target": "es2018",
+    "target": "es2019",
```
Comment (andrewrlee) [inline tsconfig.json:4]: to enable additional features like flatmap

File: package.json [hunk 2]
```diff
@@ -145,13 +145,13 @@
...
         {
-          "outputDirectory": "test_results/jest/"
+          "outputDirectory": "test-results/jest/"
         }
       ],
       [
         "./node_modules/jest-html-reporter",
         {
-          "outputPath": "test_results/unit-test-reports.html"
+          "outputPath": "test-results/unit-test-reports.html"
```
Comment (andrewrlee) [inline package.json:154]: jest and cypress were publishing to different folders - this corrects them both

File: @types/prisonApi/index.d.ts [hunk 3]
```diff
@@ -1,6866 +1,6864 @@
 /* eslint-disable camelcase */
```
Comment (bell-pa) [inline @types/prisonApi/index.d.ts:1]: I can see that you've removed a lot of code, but there still seems to be a lot of type information that BVL will never use. Do we really need it? We're only using a few end-points...

File: backend/services/bookingService.ts [hunk 4]
```diff
@@ -0,0 +1,95 @@
...
+    courtFilter: string
+  ): Promise<AppointmentResult> {
+    const [courts, bookings, locations] = await Promise.all([
+      this.whereaboutsApi.getCourtLocations(context).then(r => r.courtLocations),
+      this.whereaboutsApi.getVideoLinkBookings(context, searchDate.format('YYYY-MM-DD')),
+      this.prisonApi.getLocationsForAppointments(context, agencyId),
+    ])
+
+    const relevantBookings = bookings
+      .filter(booking => booking.agencyId === agencyId)
```
Comment (bell-pa) [inline backend/services/bookingService.ts:80]: whereabouts-api already filters by agency 'WWI'. See CourtService.kt `getVideoLinkBookingsForDateAndCourt`, line 182.

**Review by bell-pa — APPROVED** [review]
LGTM

File: @types/prisonApi/index.d.ts (hunk 3 above)
Comment (andrewrlee) [reply @types/prisonApi/index.d.ts:1]: I didn't actually remove any- I just removed a nested interface to make consuming the types easier to consume in code.

Comment (andrewrlee) [reply @types/prisonApi/index.d.ts:1]: Possibly though -I'm still not sure the trade off between making it easier to update and consume new endpoints vs having much less types to deal with

File: backend/services/bookingService.ts (hunk 4 above)
Comment (andrewrlee) [reply backend/services/bookingService.ts:80]: ah, thought it was going to be the frontends responsibility to filter. 
We can remove this though

**Pull request merged**